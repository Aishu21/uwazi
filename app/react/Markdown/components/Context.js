import React, { Component } from 'react';
import PropTypes from 'prop-types';

export const PagesContext = React.createContext('');
export default class Context extends Component {
  render() {
    const { children } = this.props;
    if (children.length) {
      return React.cloneElement(children[0], { data: this.context });
    }
    return (<PagesContext.Consumer>{value => value}</PagesContext.Consumer>);
  }
}

Context.defaultProps = {
  children: ''
};
Context.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
};
