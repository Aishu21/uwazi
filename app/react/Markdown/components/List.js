import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { PagesContext } from './Context';

class List extends Component {
  render() {
    const { children, data } = this.props;
    return (
      <React.Fragment>
        {data.map((item, index) => (
          <PagesContext.Provider value={item} key={index}>
            {React.cloneElement(children[0])}
          </PagesContext.Provider>
          ))}
      </React.Fragment>
    );
  }
}

List.defaultProps = {
  data: ['Batman', 'Robin'],
};

List.propTypes = {
  data: PropTypes.array,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};

export default List;
