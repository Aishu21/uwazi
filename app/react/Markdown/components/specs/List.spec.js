import React from 'react';

import { shallow } from 'enzyme';

import List from '../List';
import Context from '../Context';

describe('List', () => {
  let props;
  beforeEach(() => {
    props = {
      data: ['Batman', 'Spiderman']
    };
  });

  it('should render the items in data using the given html', () => {
    const component = shallow(
      <List {...props}>
        <span>Name: <Context/></span>
      </List>
    );
    expect(component).toMatchSnapshot();
  });
});
