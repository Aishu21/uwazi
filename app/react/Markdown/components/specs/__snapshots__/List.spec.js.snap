// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the items in data using the given html 1`] = `
<React.Fragment>
  <[object Object]
    key="0"
    value="Batman"
  >
    <span>
      Name: 
      <Context />
    </span>
  </[object Object]>
  <[object Object]
    key="1"
    value="Spiderman"
  >
    <span>
      Name: 
      <Context />
    </span>
  </[object Object]>
</React.Fragment>
`;
